### ISSUE TITLE
using system tree-sitter

### ISSUE THREAD

**Issue by barcharcraz** [nimterop/nimterop#274]
Is it possible to tell nimterop to use the system version of tree-sitter and avoid connecting to the internet when it first goes to generate a binding?

**Comment by genotrance** [comment]
Nimterop actually compiles in the tree-sitter library during install. It does not need tree-sitter when generating a binding.

**Comment by barcharcraz** [comment]
Yes, it downloads and builds tree-sitter when nimterop is compiled. This is a problem if you want to compile things from sources without internet access. It also makes it impossible to package nimterop (or any derived library) on most linux distributions, since they want to mirror all source code and prohibit network access during builds).

Indeed fedora has pretty much given up on packaging nim apps for this reason (along with others)

**Comment by genotrance** [comment]
I'm open to a PR that adds tree-sitter as a sub-module instead of being pulled via nimterop itself.

**Comment by barcharcraz** [comment]
that should be easy enough, hopefully I'll have time to make it happen soon.

**Comment by arkanoid87** [comment]
I'm also interested in using tree-sitter with nim for another project, it would be nice to turn the wrapping/binding part into an independent project available in nimble packages.

It would also be nice to see it available into tree-sitter github readme under "bindings"

**Comment by genotrance** [comment]
> I'm also interested in using tree-sitter with nim for another project, it would be nice to turn the wrapping/binding part into an independent project available in nimble packages.
> 
> It would also be nice to see it available into tree-sitter github readme under "bindings"

There is https://github.com/genotrance/nimtreesitter which is in Nimble. I don't know if it still works though.
